icu4j/main/classes/core/src/com/ibm/icu/impl/ZoneMeta.java

```diff
@@ -419,6 +419,38 @@ private static String findCLDRCanonicalID(String tzid) {
         return canonical;
     }
 
+    /**
+     * Returns primary IANA zone ID for the input zone ID. When input zone ID
+     * is not known, this method returns null.
+     *
+     * @param tzid  An input zone ID.
+     * @return  A primary IANA zone ID equivalent to the input zone ID.
+     */
+    public static String getIanaID(String tzid) {
+        String ianaID = null;
+        // First, get CLDR canonical ID
+        String canonicalID = getCanonicalCLDRID(tzid);
+        if (canonicalID != null) {
```

Comment:
how about we write write it in a way which has less condictional block in the function as:
```
if (canonicalID == null) return null;
try {
     UResourceBundle keyTypeData = UResourceBundle.getBundleInstance(ICUData.ICU_BASE_NAME,
                        "keyTypeData", ICUResourceBundle.ICU_DATA_CLASS_LOADER);
     UResourceBundle ianaMap = keyTypeData.get("ianaMap");
     UResourceBundle ianaTzMap = ianaMap.get("timezone");
     try {
         return ianaTzMap.getString(canonicalID.replace('/', ':'));
     } catch (MissingResourceException e) {
         // No IANA zone ID mapping. In this case, ianaId set by getCanonicalCLDRID()
         // is also a primary IANA id.
        return canonicalID;
     }
 } catch (MissingResourceException e) {
     // When CLDR canonical ID is not found for the input tzid.\
     return null;
 }
```

Reply:
done